contracts/contracts/strategies/FraxETHStrategy.sol

```diff
@@ -0,0 +1,118 @@
+
+        if (_asset == weth) {
+            // Unwrap WETH
+            IWETH9(weth).withdraw(_amount);
+
+            // Deposit ETH for frxETH and stake it
+            // slither-disable-next-line unused-return
+            fraxETHMinter.submitAndDeposit{ value: _amount }(
+                address(this)
+            );
+        } else if (_asset == address(assetToken)) {
+            // Stake frxETH
+            // slither-disable-next-line unused-return
+            IERC4626(platformAddress).deposit(_amount, address(this));
+        } else {
+            revert("Unexpected asset address");
+        }
+
+        emit Deposit(_asset, address(shareToken), _amount);
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return _asset == address(assetToken) || _asset == weth;
+    }
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        virtual
+        override
+        returns (uint256 balance)
+    {
+        if (_asset == weth) {
+            // For WETH, it's always 0
+            return 0;
+        }
+
+        // If it's not WETH, it has to be frxETH
+        require(_asset == address(assetToken), "Unexpected asset address");
+
+        /* We are intentionally not counting the amount of assetToken parked on the
+         * contract toward the checkBalance. The deposit and withdraw functions
+         * should not result in assetToken being unused and owned by this strategy
+         * contract.
+         */
+        return
+            IERC4626(platformAddress).convertToAssets(
+                shareToken.balanceOf(address(this))
+            );
+    }
+
+    /**
+     * @dev Deposit the entire balance of assetToken to gain shareToken
+     */
+    function depositAll() external virtual override onlyVault nonReentrant {
+        uint256 balance = assetToken.balanceOf(address(this));
+        if (balance > 0) {
+            _deposit(address(assetToken), balance);
+        }
+
+        // Wrap any ETH balance to WETH
+        // Wrapping here and unwrapping in `_deposit` isn't gas efficient,
+        // But `depositAll` isn't called frequently; only during reallocation.
+        uint256 ethBalance = address(this).balance;
+        if (ethBalance > 0) {
+            IWETH9(weth).deposit{ value: ethBalance }();
+        }
+
+        uint256 wethBalance = IWETH9(weth).balanceOf(address(this));
+        if (wethBalance > 0) {
+            _deposit(weth, wethBalance);
+        }
+    }
+
+    /**
+     * @dev Accept ETH
+     */
+    receive() external payable {}
```

Comment:
In case we decide our strategy doesn't want to handle eth we would remove this bit as well.

Reply:
got it thanks 👍